nitpicks and fixes and code quality and stuff

12beesinatrenchcoat opened the pull request:
a bunch of small things that'll probably add up, kinda like i did with my website. https://github.com/AndyThePie/AndyThePie.github.io/pull/16

:warning: **well, except for the fact that the project now uses xo code style (see https://github.com/AndyThePie/slabbot/commit/efe130f858f4bd0fdc3e0eea9a3da0860d3d026b). so that's pretty important.**

~~release v0.4.2 will be done after this is merged.~~ nah

12beesinatrenchcoat committed:
functions in things.functions.js are no longer named

redundant.

12beesinatrenchcoat committed:
capitalize `commandInfo.json` and move it to root

...which allows me to use "require.main.require", which is a heck of a lot cleaner, in my opinion.

12beesinatrenchcoat committed:
use "xo" code style (with modifications)

keep things a tad more consistent, eh?  exceptions are "capitalized-comments" (i like all lowercase), "no-warning-comments" (we have the todo bot for that), "radix" (i don't use anything other than decimal), and quotes (i like double quotes).

also i now use tabs. muahaha.

12beesinatrenchcoat committed:
Create things.constants.js

currently only for a consistent color in embeds.

12beesinatrenchcoat committed:
move `despacito.js` into `silly` folder + category, split videos into separate file

and just some small changes. nothing insane. for now.
and also removed cooldown (uses default).

12beesinatrenchcoat committed:
modifications to cooldowns

global default cooldown → 3000ms, cooldowns for some other commands have been significantly reduced.

12beesinatrenchcoat committed:
delete cooldown message once cooldown has expired

title says it all. resolves #33.

12beesinatrenchcoat committed:
use SLABBOT_ORANGE color for meta command embeds

keeping things constant.
also fixes a typo in help.js + gives color to help.js.

12beesinatrenchcoat committed:
square brackets for arguments

i see it more often than angled brackets, at least.

12beesinatrenchcoat committed:
Updated READMEs

more information everywhere, woot.
base README.md is also significantly better, in my opinion. and also more badges.

12beesinatrenchcoat committed:
Update .editorconfig

12beesinatrenchcoat committed:
move `toBigNumber` to `things.functions.js`

don't repeat yourself don't repeat yourself don't repeat yourself don't repeat yourself don't repeat yourself don't repeat yourself don't repeat yourself don't repeat yourself don't repeat yourself don't repeat yourself

12beesinatrenchcoat committed:
update level up message

it's now orange. and also the "LEVEL UP!" bit is in the code block.

12beesinatrenchcoat committed:
`createExpBar` end cap takes more priority

prevents | from being used when ] should be used instead.

12beesinatrenchcoat committed:
rework `me` embed

no more "compact version" (sorry mobile users!) anyways, orange.

12beesinatrenchcoat committed:
give `me` ability to get other users' info

although you have to ping them... which is... oh jeez.

AndyThePie commented:
This PR is probably getting too big, and so I'll cut it off somewhere:
This PR should resolve #36 and #23. (It already resolves #33). That's all.

12beesinatrenchcoat committed:
`commandFilter`, prevent osu command from being loaded if missing config
half-resolves #34.

12beesinatrenchcoat committed:
color `console.log` messages + log loaded commands
pretty!... i think.

12beesinatrenchcoat committed:
prevent loading of `exp`, `commandCount`, and `me` when cannot connect to mongodb
and with that, fully resolves #34!
...i might force this to only work in development environments, this shouldn't happen in production... also maybe better messaging about what's been disabled / more development environment stuff.

12beesinatrenchcoat committed:
Merge branch 'master' into nits-and-stuff
